### PR TITLE
Enable deploying a minimal set of resources for running the canary

### PR DESCRIPTION
**Pull request by jdisanti**
## Motivation and Context
This makes it possible to deploy just the minimal set of resources needed to test the canary locally against a test AWS account.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tools/ci-cdk/README.md [hunk 1]
```diff
@@ -5,6 +5,30 @@ continuous integration.
...
+npm run build
+npx cdk --app "node build/bin/canary-only.js" synth
+npx cdk --app "node build/bin/canary-only.js" deploy --outputs-file cdk-outputs.json
+```
+
+From there, you can just point the `canary-runner` to the `cdk-outputs.json` to run it:
+
+```bash
+cd canary-runner
+cargo run -- --sdk-version 0.9.0 --musl --cdk-outputs ../cdk-outputs.json
```
Comment (Velfi) [inline tools/ci-cdk/README.md:24]: Can we add a note somewhere about updating this version number on release or replacing it with something like `<version>`?

File: tools/ci-cdk/canary-runner/src/run.rs [hunk 2]
```diff
@@ -62,23 +63,82 @@ pub struct RunOpt {
...
-    /// The name of the S3 bucket to upload the canary binary bundle to
+    /// File path to a CDK outputs JSON file. This can be used instead
+    /// of all the --lambda... args.
     #[clap(long)]
-    lambda_code_s3_bucket_name: String,
+    cdk_output: Option<PathBuf>,
+
+    /// The name of the S3 bucket to upload the canary binary bundle to
+    #[clap(long, required_unless_present = "cdk-output")]
+    lambda_code_s3_bucket_name: Option<String>,
```
Comment (Velfi) [inline tools/ci-cdk/canary-runner/src/run.rs:73]: I wish Git diffs were better :(

File: tools/ci-cdk/canary-runner/src/run.rs [hunk 3]
```diff
@@ -62,23 +63,82 @@ pub struct RunOpt {
...
+                lambda_execution_role_arn: value.inner.lambda_execution_role_arn,
+            })
+        } else {
+            Ok(Options {
+                sdk_version: run_opt.sdk_version,
+                sdk_path: run_opt.sdk_path,
+                musl: run_opt.musl,
+                lambda_code_s3_bucket_name: run_opt.lambda_code_s3_bucket_name.expect("required"),
+                lambda_test_s3_bucket_name: run_opt.lambda_test_s3_bucket_name.expect("required"),
+                lambda_execution_role_arn: run_opt.lambda_execution_role_arn.expect("required"),
```
Comment (Velfi) [inline tools/ci-cdk/canary-runner/src/run.rs:131]: I feel like we should be using `clap` to do this required check. I don't have strong feelings on it though.

File: tools/ci-cdk/canary-runner/src/run.rs [hunk 4]
```diff
@@ -129,19 +189,19 @@ pub async fn run(opt: RunOpt) -> Result<()> {
     result.map(|_| ())
 }
 
-async fn run_canary(opt: RunOpt, config: &aws_config::Config) -> Result<Duration> {
+async fn run_canary(options: &Options, config: &aws_config::Config) -> Result<Duration> {
```
Comment (Velfi) [inline tools/ci-cdk/canary-runner/src/run.rs:192]: Thanks for changing it from opt to options

**Review by Velfi — APPROVED** [review]

File: tools/ci-cdk/canary-runner/src/run.rs (hunk 3 above)
Comment (jdisanti) [reply tools/ci-cdk/canary-runner/src/run.rs:131]: Clap is doing the required check already. I have to unwrap the options to get to the data in this case (but it's already validated).

**Pull request merged**